schemas/gen.py

```diff
@@ -144,9 +144,14 @@ def next_struct_name(file: str, start: int) -> str:
         for lineno, line in enumerate(f.readlines()):
             if lineno < start:
                 continue
+            # Detect struct types.
             match = re.match("type ([\\S]+) struct", line)
             if match is not None:
                 return match[1]
+            # Detect non-struct types.
+            match = re.match("type ([\\S]+)", line)
+            if match is not None:
+                return match[1]
```

Comment:
I'm confused why `next_struct_name` detects non structs. Also, this wouldn't match some structs valid according to the go spec (e.g. multiple declared in a block with ()), so maybe some docs/warning there would be good.

Reply:
Yeah, I'll rename it accordingly.